app/models/views/summary.rb

```diff
@@ -57,5 +60,14 @@ def full_address
        @applicant_address_postcode].join(' ')
     end
 
+    private
+
+    def build_total(session)
+      total = 0
+      session[:income].try(:each) do |_k, v|
+        total += v if v.present? && v.is_a?(Float)
+      end
+      "£#{total.round(0)}"
```

Comment:
I wouldn't have the currency sign in here. I'd leave that for the presentation layer, i.e. views.